gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -755,14 +773,26 @@ protected static DiffPathSet fullPathDiff(HiveLocationDescriptor sourceLocation,
     }
 
     // Now desiredTargetExistingPaths contains paths that we don't want, but which are not managed by the existing
-    // table / partition. We shouldn't delete them (they're not managed by Hive), and we don't want to pick them up
-    // in the new table / partition, so if there are any leftover files, abort copying this table / partition.
-    if (desiredTargetExistingPaths.size() > 0) {
+    // table / partition.
+    // Ideally, we shouldn't delete them (they're not managed by Hive), and we don't want to pick
+    // them up in the new table / partition, so if there are any leftover files, we should abort copying
+    // this table / partition.
+    if (desiredTargetExistingPaths.size() > 0 && helper.getManagedDataPolicy() != ManagedDataPolicy.DELETE_UNMANAGED_DATA) {
       throw new IOException(String.format(
           "New table / partition would pick up existing, undesired files in target file system. " + "%s, files %s.",
-          partition.isPresent() ? partition.get().getCompleteName() : helper.dataset.table.getCompleteName(),
+          partition.isPresent() ? partition.get().getCompleteName() : helper.getDataset().getTable().getCompleteName(),
           Arrays.toString(desiredTargetExistingPaths.keySet().toArray())));
     }
+    // Unless, the policy requires us to delete such un-managed files - in which case: we will add the leftover files
+    // to the deletion list.
+    else if (desiredTargetExistingPaths.size() > 0) {
+      log.warn("Un-managed files detected in target file system, however deleting them because of the policy: " +
+        ManagedDataPolicy.DELETE_UNMANAGED_DATA);
+      for (Path delete : desiredTargetExistingPaths.keySet()) {
+        log.warn("Un-managed file to be deleted: " + delete);
```

Comment:
I would prefer not to log every file in a separate line. Can we do a single line with all files to be deleted?